superset/models/helpers.py

```diff
@@ -1853,14 +1853,21 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                     if op == utils.FilterOperator.NOT_IN.value:
                         cond = ~cond
                     where_clause_and.append(cond)
-                elif op == utils.FilterOperator.IS_NULL.value:
-                    where_clause_and.append(sqla_col.is_(None))
-                elif op == utils.FilterOperator.IS_NOT_NULL.value:
-                    where_clause_and.append(sqla_col.isnot(None))
+                elif op in {
+                    utils.FilterOperator.IS_NULL.value,
+                    utils.FilterOperator.IS_NOT_NULL.value,
```

Comment:
Cast `op` to an enum earlier with (though you need to change the variable name to prevent mypy from complaining the type has changed):

```python
try:
    op = utils.FilterOperator(op)
except ValueError:
    valid = ", ".join(operator.value for operator in utils.FilterOperator)
    raise SomeException(f"Invalid operator type: {op}. Supported values are: {valid}")
```

Then do all comparisons with the enum instead of string:

```suggestion
                elif op in {
                    utils.FilterOperator.IS_NULL,
                    utils.FilterOperator.IS_NOT_NULL,
```